.github/workflows/nixf-tidy-action.yaml

```
@@ -0,0 +1,25 @@
+name: nixf-tidy code linter
+
+on:
+  [ pull_request ]
```

Prefer YAML syntax to JSON-style

Also, I'm not sure [pull_request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) is the right event here. That runs whenever a PR is opened, edited, reviewed, etc (see "activity types").

Instead, we'd want this to run whenever a branch is updated ([push](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push)).

We _could_ use `branches-ignore` to prevent running on `main` and `nixos-*`, but IMO I don't see an issue with having the CI on these branches too.

Note if the action expects `github.event.pull_request.head.sha`, this'll be missing when using `on.push`. Instead `github.sha` or `GITHUB_SHA` should be used for non-pr events.

```suggestion
  push
```

---

> Also, I'm not sure [pull_request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) is the right event here. That runs whenever a PR is opened, edited, reviewed, etc (see "activity types"). 

I haven't test it exactly for events other than "pull_request", maybe I should update the action itself then.

---

> Also, I'm not sure [pull_request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) is the right event here. That runs whenever a PR is opened, edited, reviewed, etc (see "activity types").

I think some of my assumptions were off here. The `pull_request` event does get triggered by a bunch of stuff, but pushes to a PR branch don't _always_ seem to trigger the `push` event...